File: day16/ans.cpp
// This is something I didn't get. This is borrowed from the solution presented
class Solution {
public:
    bool checkValidString(string s) {
        int low = 0;
        int high = 0;
        
        for(auto ch: s) {
            if(ch == '(') {
                low += 1;
            } else {
                low -= 1;
            }
            
            if(ch != ')') {
                high += 1;
            } else {
                high -= 1;
            }
            
            if(high < 0) {
                return false;
            }
            
            low = max(0, low);
        }
        
        return low == 0;
    }
};

File: day19/ans.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low = 0
        high = len(nums) - 1
        
        while low <= high:
            middle = low + ((high - low) // 2)
            
            if nums[middle] == target:
                return middle
            
            if nums[low] <= nums[middle]:
                if target >= nums[low] and target < nums[middle]:
                    high = middle - 1
                else:
                    low = middle + 1
            else:
                if target >= nums[low] or target < nums[middle]:
                    high = middle - 1
                else:
                    low = middle + 1
                
        return -1
        
